ui/apps/platform/src/Containers/NetworkGraph/NetworkGraphPage.tsx

```diff
@@ -101,6 +108,26 @@ function NetworkGraphPage() {
     const selectedCluster = { name: clusterFromUrl, id: selectedClusterId };
     const { deploymentCount } = useFetchDeploymentCount(selectedClusterId || '');
 
+    const [prevEpochCount, setPrevEpochCount] = useState(0);
+    const [currentEpochCount, setCurrentEpochCount] = useState(0);
+
+    // We will update the poll epoch after 30 seconds to update the node count for a cluster
+    useInterval(() => {
+        setPollEpoch(pollEpoch + 1);
+    }, 30000);
```

Comment:
I'm curious about the use of interval vs. timeout. In this scenario we could have this possibility:
1. Interval updates
2. API is triggered
3. API still hasn't returned within 30 seconds due to possible network slowdown
4. Interval updates

Would it be better to apply a timeout to call the API again after a successful response rather than use an interval?

Reply:
While a timeout that re-initializes itself would be safer for short time periods, API calls that don't return within 30 seconds are indicative of a larger problem.

We are adopted this useInterval hook in seven other sections of the app that poll, to replace the redux-saga version of polling, such as in this case `yield call(delay, 30000);`